ext/mbstring/php_mbregex.c

```diff
@@ -1400,6 +1400,11 @@ PHP_FUNCTION(mb_ereg_search_setpos)
 		return;
 	}
 
+	/* Accept negative position if length of search string can be determined */
+	if ((position < 0) && (!Z_ISUNDEF(MBREX(search_str))) && (Z_TYPE(MBREX(search_str)) == IS_STRING)) {
+		position += Z_STRLEN(MBREX(search_str));
```

Comment:
If we store the negative value, everyone reading it must be able to handle negative values, including mb_ereg_search_getpos(). If this function is called when we don't have a search string, what should it return ? the negative value ? Accepting negative value when search string is defined only and computing the 'real' offset immediately is not perfect but storing a negative value when every existing code is expecting the value to be positive or null has implications I'm not sure to master.